cmd/main.go

```diff
@@ -79,13 +70,23 @@ func main() {
 	// listen concurrently to allow for graceful shutdown
 	go func() {
 		log.Infof(nil, "Service Revision %s built on %s", configuration.Commit, configuration.BuildTime)
-		log.Infof(nil, "Listening on %q...", srv.Config().GetHTTPAddress())
+		log.Infof(nil, "Listening on %q...", configuration.HTTPAddress)
 		if err := srv.HTTPServer().ListenAndServe(); err != nil {
 			log.Error(nil, err, err.Error())
 		}
 	}()
 
-	gracefulShutdown(srv.HTTPServer(), srv.Config().GetGracefulTimeout())
+	// update cache every 10 seconds
+	go func() {
+		for {
+			if _, err := commonconfig.ForceLoadToolchainConfig(cl); err != nil {
+				log.Error(nil, err, "failed to update the configuration cache")
+			}
+			time.Sleep(10 * time.Second)
+		}
+	}()
```

Comment:
I usually use the `Until` function provided by `k8s.io/apimachinery/pkg/util/wait` for such use-cases:
```
go wait.Until(func() {
		...
	}, period, stopChannel)
``` 
but your version is fine too - just wanted to share another solution.